From 20811c5aa9c528405eaa3f88ff22a9adebafa9e6 Mon Sep 17 00:00:00 2001
From: Nicolas Cavallari <nicolas.cavallari@green-communications.fr>
Date: Wed, 8 Mar 2023 14:00:52 +0100
Subject: [PATCH] sysdir: Do not declare win32 functions on non-win32 platforms

These declaration poses problems on some embedded or retro Linux systems
that deliberately disable support for wchar_t from their libc.

Signed-off-by: Nicolas Cavallari <nicolas.cavallari@green-communications.fr>
[Retrieved from:
https://github.com/libgit2/libgit2/commit/20811c5aa9c528405eaa3f88ff22a9adebafa9e6]
---
 src/libgit2/sysdir.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libgit2/sysdir.h b/src/libgit2/sysdir.h
index 1d15bbf43..03f59e1de 100644
--- a/src/libgit2/sysdir.h
+++ b/src/libgit2/sysdir.h
@@ -134,10 +134,12 @@ extern int git_sysdir_set(git_sysdir_t which, const char *paths);
  */
 extern int git_sysdir_reset(void);
 
+#ifdef GIT_WIN32
 /** Sets the registry system dir to a mock; for testing.  */
 extern int git_win32__set_registry_system_dir(const wchar_t *mock_sysdir);
 
 /** Find the given system dir; for testing. */
 extern int git_win32__find_system_dirs(git_str *out, const char *subdir);
+#endif
 
 #endif
-- 
2.39.2

